compiler/src/dotty/tools/dotc/transform/ElimByName.scala

```diff
@@ -80,7 +80,9 @@ class ElimByName extends MiniPhaseTransform with InfoTransformer { thisTransform
             val inSuper = if (ctx.mode.is(Mode.InSuperCall)) InSuperCall else EmptyFlags
             val meth = ctx.newSymbol(
                 ctx.owner, nme.ANON_FUN, Synthetic | Method | inSuper, MethodType(Nil, Nil, argType))
-            Closure(meth, _ => arg.changeOwner(ctx.owner, meth))
+            atGroupEnd { implicit ctx: Context =>
+              Closure(meth, _ => arg.changeOwner(ctx.owner, meth))
```

Comment:
try `changeOwnerAfter`. It's supposed to fix the issue. No need for `atGroupEnd`.